### ISSUE TITLE
Pylance keeps saying false reportGeneralTypeIssues

### ISSUE THREAD

**Issue by BGOPC** [microsoft/pyright#5170]
**Describe the bug**
Pylance keeps giving me false #reportGeneralTypeIssues in any part of the code, no library imported or etc, just std python and i keep getting this kind of false error

**To Reproduce**
This code is given an error in the last line by Pylance while it runs perfectly
```Python
string = "fhdshdsgfs"
word = ""
for j in range(len(part)):
            if j != 0:
                word += "and"
            word += ones.get(part[j])
```

**Expected behavior**
No error gets shown by Pylance

**Screenshots or Code**
If applicable, could you add screenshots or the text of the code (surrounded by triple backticks) to help explain your problem?
![image](https://github.com/microsoft/pyright/assets/94793764/d64aa17f-b7c6-4771-8746-68c7dc9dcdcb)3
This image is for the code above

**VS Code extension or command-line**
VsCode extension the `5/18/2023` release

**Additional context**
This happens almost every time and it's not just this specific piece of code

**Comment by erictraut** [comment]
Could you provide a minimal, self-contained code sample that exhibits the problem? The code sample that you've posted above references symbols that are not defined (`part` and `ones`). Without information about how these symbols are defined, it's not possible to repro the issue you're seeing.

**Comment by BGOPC** [comment]
Ofc, Here you go:
```python
ones = {
        "1": "one",
        "2": "two",
        "3": "three",
        "4": "four",
        "5": "five",
        "6": "six",
        "7": "seven",
        "8": "eight",
        "9": "nine",
        "10": "ten",
        "11": "eleven",
        "12": "twelve",
        "13": "thirteen",
        "14": "fourteen",
        "15": "fifteen",
        "16": "sixteen",
        "17": "seventeen",
        "18": "eighteen",
        "19": "nineteen",
        "20": "twenty"
    }

part = "fhdshdsgfs"
word = ""
for j in range(len(part)):
    if j != 0:
        word += "and"
        word += ones.get(part[j])
```

**Comment by erictraut** [comment]
The error is caused by the fact that the `get` method for a `dict` object can return `None` if the key is not found in the dictionary. If you right click on `get` and choose "Go To Declaration", you can see the type declaration for `get`. This type declaration comes from typeshed, which is a community-maintained collection of type stub files that describe how the Python stdlib works from a type perspective.

In your particular code, it can be seen through examination that you are accessing keys that are always present, but a static type checker won't be able to determine that.

You have a few options:
1. If you don't find these type checking errors helpful in general, you can disable them for your project by setting "python.analysis.typeCheckingMode" to "off".
2. You can suppress this single error by adding a `# type: ignore` comment to the end of that line.
3. You can modify your code to make it clear that `get` never returns `None` by supplying a default value: `ones.get(part[j], '')`.

In any case, pylance is working as designed here, so this isn't considered a bug.
